pkg/skaffold/build/builder_mux.go

```diff
@@ -65,20 +66,29 @@ func NewBuilderMux(cfg Config, store ArtifactStore, builder func(p latestV1.Pipe
 
 		if cfg.BuildConcurrency() >= 0 {
 			minConcurrency = cfg.BuildConcurrency()
-		} else {
-			concurrency := b.Concurrency()
-			// set mux concurrency to be the minimum of all builders' concurrency. (concurrency = 0 means unlimited)
-			switch {
-			case minConcurrency < 0:
-				minConcurrency = concurrency
-				log.Entry(context.TODO()).Infof("build concurrency first set to %d parsed from %s[%d]", minConcurrency, reflect.TypeOf(b).String(), i)
-			case concurrency > 0 && (minConcurrency == 0 || concurrency < minConcurrency):
-				minConcurrency = concurrency
-				log.Entry(context.TODO()).Infof("build concurrency updated to %d parsed from %s[%d]", minConcurrency, reflect.TypeOf(b).String(), i)
-			default:
-				log.Entry(context.TODO()).Infof("build concurrency value %d parsed from %s[%d] is ignored since it's not less than previously set value %d", concurrency, reflect.TypeOf(b).String(), i, minConcurrency)
-			}
+			continue
 		}
+
+		if b.Concurrency() == nil {
+			continue
+		}
```

Comment:
This doesn't seem to be equivalent.  Consider if I have two modules, A and B.  Both have local build-envs.  A does not specify a build concurrency.  B specified concurrency 0.  I believe that prior to this change, A would have had a default concurrency of 1 set which would have resulted in the minConcurrency==1 (serialized).  Whereas now A would be skippedI think we will have minConcurrency 0.

Honestly, I think we should separate this out into a function whose results can be tested for different scenarios.

Reply:
@briandealwis I verified via this test [here](https://github.com/GoogleContainerTools/skaffold/pull/7182/commits/f8904bd6575a22bca86fa7bb4c078fe13d9ea3b2#diff-1dd11874c1929244aebdbb3fe85c5120f832b2c8b17f53be4276af55a0dce90aR122) that if B specified concurrency 0 and A does not set the concurrency, then it default to 1.